src/mutation/RelayMutationQueue.js

```diff
@@ -350,14 +355,15 @@ class PendingTransaction {
 
   getCollisionKey(): ?string {
     if (this._collisionKey === undefined) {
-      this._collisionKey = this.mutation.getCollisionKey() || null;
+      this._collisionKey =
+        this.mutation.getCollisionKey(this.getProps()) || null;
     }
     return this._collisionKey;
   }
 
   getConfigs(): Array<RelayMutationConfig> {
     if (!this._configs) {
-      this._configs = this.mutation.getConfigs();
+      this._configs = this.mutation.getConfigs(this.getProps());
```

Comment:
In the description to #699 I tried to explain why I am passing props to `RelayMutation` methods. But if you know a better solution I will be happy to change the PR.